drivers/i2s/i2s_mcux_flexcomm.c

```diff
@@ -0,0 +1,899 @@
+		fsl_cfg->wsPol = true;
+		break;
+	case I2S_FMT_DATA_FORMAT_RIGHT_JUSTIFIED:
+		fsl_cfg->mode = kI2S_ModeDspWs50;
+		fsl_cfg->wsPol = true;
+		break;
+	default:
+		LOG_ERR("Unsupported I2S data format");
+		return -EINVAL;
+	}
+
+	if (fsl_cfg->masterSlave == kI2S_MasterSlaveNormalMaster ||
+		fsl_cfg->masterSlave == kI2S_MasterSlaveWsSyncMaster) {
+		fsl_cfg->divider = base_frequency /
+				   i2s_cfg->frame_clk_freq /
+				   fsl_cfg->frameLength;
+	}
+
+	/*
+	 * Set frame and bit clock polarity according to
+	 * inversion flags.
+	 */
+	switch (i2s_cfg->format & I2S_FMT_CLK_FORMAT_MASK) {
+	case I2S_FMT_CLK_NF_NB:
+		break;
+	case I2S_FMT_CLK_NF_IB:
+		fsl_cfg->sckPol = !fsl_cfg->sckPol;
+		break;
+	case I2S_FMT_CLK_IF_NB:
+		fsl_cfg->wsPol = !fsl_cfg->wsPol;
+		break;
+	case I2S_FMT_CLK_IF_IB:
+		fsl_cfg->sckPol = !fsl_cfg->sckPol;
+		fsl_cfg->wsPol = !fsl_cfg->wsPol;
+		break;
+	default:
+		LOG_ERR("Unsupported clocks polarity");
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static struct i2s_config *i2s_mcux_config_get(const struct device *dev,
+				enum i2s_dir dir)
+{
+	struct i2s_mcux_data *dev_data = dev->data;
+	struct stream *stream;
+
+	if (dir == I2S_DIR_RX) {
+		stream = &dev_data->rx;
+	} else {
+		stream = &dev_data->tx;
+	}
+
+	if (stream->state == I2S_STATE_NOT_READY) {
+		return NULL;
+	}
+
+	return &stream->cfg;
+}
+
+static int i2s_mcux_configure(const struct device *dev, enum i2s_dir dir,
+			       struct i2s_config *i2s_cfg)
+{
+	const struct i2s_mcux_config *cfg = dev->config;
+	struct i2s_mcux_data *dev_data = dev->data;
+	I2S_Type *base = cfg->base;
+	struct stream *stream;
+	uint32_t base_frequency;
+	i2s_config_t fsl_cfg;
+	int result;
+	uint8_t data_len = 0;
+	uint8_t bytes_per_word = 0;
+
+	if (dir == I2S_DIR_RX) {
+		stream = &dev_data->rx;
+	} else if (dir == I2S_DIR_TX) {
+		stream = &dev_data->tx;
+	} else {
+		LOG_ERR("Either RX or TX direction must be selected");
+		return -EINVAL;
+	}
+
+	if (stream->state != I2S_STATE_NOT_READY &&
+	    stream->state != I2S_STATE_READY) {
+		LOG_ERR("invalid state");
+		return -EINVAL;
+	}
+
+	if (i2s_cfg->frame_clk_freq == 0U) {
+		stream->state = I2S_STATE_NOT_READY;
+		return 0;
+	}
+
+	/* Figure out function base clock */
+	if (clock_control_get_rate(cfg->clock_dev,
+				   cfg->clock_subsys, &base_frequency)) {
+		return -EINVAL;
+	}
+
+	/*
+	 * Validate the configuration by converting it to SDK
+	 * format.
+	 */
+	result = i2s_mcux_flexcomm_cfg_convert(base_frequency, dir, i2s_cfg,
+					       &fsl_cfg);
+	if (result != 0) {
+		return result;
+	}
+
+	/* Apply the configuration */
+	if (dir == I2S_DIR_RX) {
+		I2S_RxInit(cfg->base, &fsl_cfg);
+	} else {
+		I2S_TxInit(cfg->base, &fsl_cfg);
+	}
+
+	/* Data length in bits */
+	data_len = (uint8_t)(((base->CFG1 & I2S_CFG1_DATALEN_MASK) >>
+						I2S_CFG1_DATALEN_SHIFT) + 1U);
+
+	/* Convert to bytes */
+	if (data_len % 8) {
+		bytes_per_word = (data_len / 8) + 1;
+	} else {
+		bytes_per_word = data_len / 8;
+	}
```

Comment:
This works, or you could use a little bit simpler implementation as in [get_word_size_bytes()](https://github.com/zephyrproject-rtos/zephyr/blob/c5caa9b916d15212228116a467ee6f5ccdb519ee/drivers/i2s/i2s_sam_ssc.c#L492)

Reply:
Thanks for pointing this, I have updated the implementation based on your suggestion.